### PR TITLE
chore(uptime): Refactor `create_project_uptime_subscription` to `get_or_create_project_uptime_subscription`

### PR DESCRIPTION
**Pull request by wedamija**
As part of implementing the update api, I want to change this interface to handle everything around creating a subscription. Splitting out this refactor

### CODE REVIEW

File: src/sentry/uptime/models.py [hunk 1]
```diff
@@ -154,6 +154,9 @@ def get_audit_log_data(self):
             "url": self.uptime_subscription.url,
             "interval_seconds": self.uptime_subscription.interval_seconds,
             "timeout": self.uptime_subscription.timeout_ms,
+            "method": self.uptime_subscription.method,
+            "headers": self.uptime_subscription.headers,
+            "body": self.uptime_subscription.body,
```
Comment (evanpurkhiser) [inline src/sentry/uptime/models.py:159]: unrelated?

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/uptime/models.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/uptime/models.py:159]: Yeah, something we missed when adding these new fields, just going to include it here
